### PR TITLE
Convert auth helpers to a use case class

### PR DESCRIPTION
**Pull request by samtgarson**
This PR makes two changes:

- Converts the auth helpers into a Usecase (this means all authentication related code should be cleanly separated from the rest of the codebase, derisking future changes)
- Only verifies the JWT token in staging/prod (this removes the need for developers to have the Hackney JWT token on their local machines, again reducing surface area for security risk)

### CODE REVIEW

File: src/pages/_app.test.tsx [hunk 1]
```diff
@@ -25,95 +23,109 @@ MockedNextApp.getInitialProps.mockImplementation(async () => pageProps);
...
-        const result = await CustomApp.getInitialProps(clientSideContext);
-
-        expect(result).toEqual({ accessDenied: true });
-        expect(replaceSpy).toHaveBeenCalledWith(`/login?redirect=%2Fpath`);
+      describe('on server side', () => {
+        beforeEach(async () => {
+          result = await CustomApp.getInitialProps(serverContext);
+        });
+
+        it('calls executes the authorizer correctly', () => {
```
Comment (bogdan-zaharia) [inline src/pages/_app.test.tsx:67]: I think either "calls" or "executes" should be removed

File: src/pages/_app.test.tsx [hunk 2]
```diff
@@ -25,95 +23,109 @@ MockedNextApp.getInitialProps.mockImplementation(async () => pageProps);
...
-        pathIsWhitelisted.mockImplementation(() => true);
-        const result = await CustomApp.getInitialProps(appContext);
-
-        expect(result).toEqual(pageProps);
+      describe('on client side', () => {
+        beforeEach(async () => {
+          result = await CustomApp.getInitialProps(clientContext);
+        });
+
+        it('calls executes the authorizer correctly', () => {
```
Comment (bogdan-zaharia) [inline src/pages/_app.test.tsx:92]: Here as well

File: src/services/request-authorizer.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+  // private deleteSession(res: ServerResponse): void => {
+  //   res.setHeader(
+  //     'Set-Cookie',
+  //     cookie.serialize(cookieName, '', {
+  //       maxAge: -1,
+  //       domain: '.hackney.gov.uk',
+  //     })
+  //   );
+  // };
```
Comment (bogdan-zaharia) [inline src/services/request-authorizer.ts:110]: Should this be removed?

**Review by bogdan-zaharia — COMMENTED** [review]
LGTM 👍, left a couple comments

File: src/services/request-authorizer.ts (hunk 3 above)
Comment (samtgarson) [reply src/services/request-authorizer.ts:110]: I think it's a useful snippet for when we need to do logout

**Review by bogdan-zaharia — APPROVED** [review]

**Pull request merged**